src/main/scala/dev/kamu/cli/external/DockerClient.scala

```diff
@@ -107,12 +110,22 @@ class DockerClient(fileSystem: FileSystem) {
       .exitValue()
   }
 
+  // give a container t seconds to terminate, kill otherwise. 10 seconds is docker's default
+  def stop(container: String, time: Int = 10): Unit = {
+    val processBuilder = prepare(
+      Seq("docker", "stop", s"--time=$time", container)
+    )
+    processBuilder
+      .run(IOHandlerPresets.logged(logger))
+      .exitValue()
+  }
+
   def pull(image: String): Unit = {
     prepare(Seq("docker", "pull", image)).!
   }
 
   def inspectHostPort(container: String, port: Int): Option[Int] = {
-    val format = "--format={{ (index (index .NetworkSettings.Ports \"" + port + "/tcp\") 0).HostPort }}"
+    val format = "--format=\'{{ (index (index .NetworkSettings.Ports \"" + port + "/tcp\") 0).HostPort }}\'"
```

Comment:
This is the change that causes tests to fail.

Quotes are necessary to run such command in the shell because shell has to split the input string into process arguments, but here the list of arguments is passed directly into the system call and extra quotes **become** part of an argument.

If your goal was to make this log a command that can be easily copy-pasted into a shell then argument quoting/escaping should be only added for the log output.